bodo/io/stream_parquet_write.py

```diff
@@ -46,13 +46,27 @@
 PARQUET_WRITE_CHUNK_SIZE = int(256e6)
 
 
-class ParquetWriterType(types.Type):
+class ParquetWriterType(StreamingStateType):
     """Data type for streaming Parquet writer's internal state"""
 
     def __init__(self, input_table_type=types.unknown):
         self.input_table_type = input_table_type
         super().__init__(name=f"ParquetWriterType({input_table_type})")
 
+    def is_precise(self):
+        return self.input_table_type != types.unknown
+
+    def unify(self, typingctx, other):
+        """Unify two ParquetWriterType instances when one doesn't have a resolved
+        input_table_type.
+        """
+        if isinstance(other, ParquetWriterType):
+            if not other.is_precise() and self.is_precise():
+                return self
+
+            # Prefer the new type in case append table changed its table type
+            return other
+
```

Comment:
Same duplication comment as above.  Just type is differing so abstract it?

Reply:
Same as above.